### ISSUE TITLE
Couple of Roll-Ups not working

### ISSUE THREAD

**Issue by HarryRSamuel** [afawcett/declarative-lookup-rollup-summaries#390]
I've created a whole load of roll-ups and there are two of them which don't appear to be working. I'm creating them as 2 separate issues. The first one is the field "Current Actual Audience Reach" which is on a custom object "Application". I want this to be pulled automatically from a field of the same name on the custom object "Live Event":

![image](https://cloud.githubusercontent.com/assets/20912161/18018858/1b921c66-6bd1-11e6-8219-aa8735ab3eaf.png)

I have a very similar field, "Current Actual User Acquisition", which seems to be working fine:

![image](https://cloud.githubusercontent.com/assets/20912161/18018886/3d3cc5e6-6bd1-11e6-8396-9f31c2741fdf.png)

I've tried manually clicking Calculate several times to no avail. On the Application object I'm seeing most of the data working fine but not this field:

![image](https://cloud.githubusercontent.com/assets/20912161/18018921/69953268-6bd1-11e6-81ee-49054a885f31.png)

The Live Event it's drawing from is showing the following info and it's the only record related to the Application in question so the amounts should match exactly:

![image](https://cloud.githubusercontent.com/assets/20912161/18018948/96eb7114-6bd1-11e6-86d7-a6c23b908a28.png)

**Comment by afawcett** [comment]
Everything looks Ok from what you have shown. Keep in mind the tool only recalcs the rollup on insert, delete and update, but in the update case ONLY if the value of the fields listed in Field to Aggregate, Relationship Criteria Fields or Relationship Field have changed. When your testing are you modifying these fields? Which specific scenarios, create, update or delete is it not firing on? Also if you only have this rollup active does it still not work?

**Comment by HarryRSamuel** [comment]
I see, I wasn't changing those fields - when I do change those fields then it works.

There is another object, Event Date, which is the child object of Live Event, and the Current Actual Audience Reach for Live Event is aggregated from the same field on Event Dates using a further roll-up based on two time-based formula checkboxes - Event in Future = FALSE and In Current Reporting Period = TRUE. Interestingly, if I change the Date & Time field on an Event Date, the Live Event does recalculate (which in turn triggers the Application to recalculate accordingly).

However, based on what you said in the other thread about it not recalculating when a formula field's value changes I'm not sure why this is working. I'm not complaining of course, but it would be useful to understand why so I know it's not a fluke!

![image](https://cloud.githubusercontent.com/assets/20912161/18090246/b74c01ac-6ebb-11e6-92ac-0c8de7917115.png)

**Comment by afawcett** [comment]
Yeah as far as i know changes in formula fields cannot be detected in triggers, is it possible the Audience_Reach__c value is changing and its that that is causing the recalc?

**Comment by wes1278** [comment]
Agreed with @afawcett. But if your formula is only a calculation of fields on the records your care about, dlrs will work. Just add all dependent fields to the relationship criteria fields data point of the rollups record

**Comment by HarryRSamuel** [comment]
The only fields which should be changing when I alter the Date & Time are the two formula fields specified in the criteria, so that must somehow be what's triggering the recalculation. However if I leave the Date & Time the same and the value of those formulae change due to the passage of time, presumably that won't trigger it.

**Comment by HarryRSamuel** [comment]
I've just tested the above and sure enough as you said, the passage of time doesn't trigger a recalc. This is not an issue for Orders because those are changing so often that they will be triggered anyway, however for Live Events and Event Dates there is a danger that some data will be left out at the end of each month.

For that reason I've changed the recalc on Audience Reach & User Acquisition for the roll-up from Event Dates to Live Events to scheduled hourly. I've left the roll-up for the same fields which then roll up from Live Events to Applications as realtime, on the basis that the scheduled recalc will update those fields on Live Events and that will consequently trigger the realtime recalc on Applications upstream. Let me know if that's a faulty assumption, otherwise I think I've got this sorted for now.
